src/Server.kt

```diff
@@ -197,16 +199,39 @@ fun Application.module() {
 
                     if (call.parameters.contains("simulationIndex")) {
                         val simIndex = call.parameters["simulationIndex"]?.toInt() ?: 0
-                        call.respondText(model.run(simIndex))
+                        call.respondText(model.run(simIndex, null))
                     } else {
-                        call.respondText(model.run(0))
+                        call.respondText(model.run(0, null))
                     }
                 } catch (err: IndexOutOfBoundsException) {
                     call.respond(HttpStatusCode.NotFound)
                 }
             }
 
         }
+
+        post("/execute/{i}") {
+            val sim = call.receive<FskSimulation>()
+            //println("SERVER: Message from the client: $sim");
+            call.parameters["i"]?.toInt()?.let {
+                try {
+                    val modelFile = modelFiles[it]
+                    val model = readModel(modelFile)
+
+                    call.respondText(model.run(0, sim))
```

Comment:
If the run method is simplified with a single *simulation* parameter the model.run call can be simplified by passing only *sim*.

Reply:
done